### ISSUE TITLE
Improve support of Reference-only and outpainting for SDXL

### ISSUE THREAD

**Issue by patrickvonplaten** [huggingface/diffusers#4257]
Reference-only has shown be a very powerful mechanism for outpainting as well as image variation. We should investigate a bit how we can best support this in a modularized, library-friendly way in `diffusers`. Also I think we should try this out for SDXL.

Related links:
- https://github.com/Mikubill/sd-webui-controlnet/discussions/1280
- https://github.com/Mikubill/sd-webui-controlnet/discussions/1464
- https://github.com/Mikubill/sd-webui-controlnet/discussions/1236

and 

https://github.com/huggingface/diffusers/blob/main/examples/community/stable_diffusion_controlnet_reference.py
https://github.com/huggingface/diffusers/blob/main/examples/community/stable_diffusion_reference.py

@DN6 maybe this could be a first fun project for you

**Comment by bghira** [comment]
SDXL 1.5 or SD 1.5? I got very excited for a moment, but now I'm not sure :sweat_smile:

**Comment by patrickvonplaten** [comment]
SD XL sorry :sweat_smile:

**Comment by yamkz** [comment]
I have always wanted this feature. I wish it could be included in the diffuser library, with support for SDXL.

**Comment by yanchaoguo** [comment]
+1

**Comment by DN6** [comment]
I think this PR: #4502 is relevant. SDXL reference pipeline will be added to Community Pipelines

**Comment by sayakpaul** [comment]
@DN6 could you maybe jam on #4502 then to help get it merged? But I concur with Patrick that we should experiment with this ourselves and make it a part of the core library.

**Comment by VigneshHexo** [comment]
> Reference-only has shown be a very powerful mechanism for outpainting as well as image variation

This one @sayakpaul

**Comment by DN6** [comment]
SDXL Reference is now available as a Community Contributed Pipeline in #4502. Opened a separate ticket for outpainting: #4718. 

Closing this issue for now.

**Comment by patrickvonplaten** [comment]
I think we should dive a bit deeper here and run some experiments. It would be really nice to have a fully working outpainting workflow for SDXL

**Comment by patrickvonplaten** [comment]
Also cc @williamberman maybe inpainting can help here as well

**Comment by VigneshHexo** [comment]
> I think we should dive a bit deeper here and run some experiments. It would be really nice to have a fully working outpainting workflow for SDXL

Will be very happy to help with this! If there is a specific to do list, will pick it up from there and get it done! Otherwise if someone is working on this, would be very happy to tag along and help them out! Please let me know @patrickvonplaten

**Comment by VigneshHexo** [comment]
> I think we should dive a bit deeper here and run some experiments. It would be really nice to have a fully working outpainting workflow for SDXL

Spoke to @sayakpaul regarding this. @DN6, @williamberman  Will be very happy to help with this! If there is a specific to do list, will pick it up from there and get it done! Please let me know! Thank you very much

**Comment by patrickvonplaten** [comment]
Essentially doing a more in-depth study of how SDXL can be used for reference-only would be very helpful (with attached pictures)

**Comment by VigneshHexo** [comment]
Will do it and get back Patrick!

**Comment by DN6** [comment]
Just some notes on the current state of Outpainting for SDXL 
https://github.com/huggingface/diffusers/issues/4718#issue-1861853770

**Comment by kelisiya** [comment]
can you support inference_only + lama inpainting like webui ?

**Comment by joyennc** [comment]
The new reference only lacks the ability to effectively utilize reference images to generate new perspectives, especially side and back views. I hope the administrator ccould take note of this and look forward to reference only continuous updates.thanks so much！～

**Comment by patrickvonplaten** [comment]
@DN6 I think it would still make sense to dive a bit into the reference pipelines trying to gauge whether they are a useful addition to core diffusers

**Comment by DN6** [comment]
@patrickvonplaten Wouldn't the IP-Adapter accomplish similar functoinality? 
https://github.com/huggingface/diffusers/pull/4944#issuecomment-1796294139

**Comment by patrickvonplaten** [comment]
Yeah maybe let's go with this first and see if we could make a 1-to-1 comparison somehow

**Comment by sayakpaul** [comment]
@asomoza is this still relevant?
